app/components/explore/side-bar.js

```diff
@@ -16,9 +26,57 @@ export default Component.extend({
     },
 
     dateValidate(date) {
-      if (moment(date).isAfter(this.get('endDate'))) {
-        this.set('endDate', date);
+      if (moment(date).isAfter(this.get('customEndDate'))) {
+        this.set('customEndDate', date);
       }
+
+      this.send('selectDateFilter', 'custom_dates');
+    },
+
+    selectDateFilter(dateType) {
+      let newStartDate = null;
+      let newEndDate = null;
+
+      if (dateType === this.get('dateType') && dateType !== 'custom_dates') {
+        this.set('dateType', null);
+      } else if (dateType === 'custom_dates') {
+        this.set('dateType', dateType);
+        newStartDate = this.get('customStartDate');
+        newEndDate = this.get('customEndDate');
+      } else if (dateType === 'all_dates') {
+        this.set('dateType', dateType);
+      } else if (dateType === 'today') {
+        this.set('dateType', dateType);
+        newStartDate = moment().toISOString();
+        newEndDate = moment().toISOString();
+      } else if (dateType === 'tomorrow') {
+        this.set('dateType', dateType);
+        newStartDate = moment().add(1, 'day').toISOString();
+        newEndDate = newStartDate;
+      } else if (dateType === 'this_week') {
+        this.set('dateType', dateType);
+        newStartDate = moment().startOf('week').toISOString();
+        newEndDate = moment().endOf('week').toISOString();
+      } else if (dateType === 'this_weekend') {
+        this.set('dateType', dateType);
+        newStartDate = moment().isoWeekday('Friday').toISOString();
+        newEndDate = moment().isoWeekday('Sunday').toISOString();
+      } else if (dateType === 'next_week') {
+        this.set('dateType', dateType);
+        newStartDate = moment().isoWeekday('Monday').add(1, 'week').toISOString();
+        newEndDate = moment().isoWeekday('Sunday').add(1, 'week').toISOString();
+      } else if (dateType === 'this_month') {
+        this.set('dateType', dateType);
+        newStartDate = moment().startOf('month').toISOString();
+        newEndDate = moment().endOf('month').toISOString();
+      }
```

Comment:
Except the first `if` statement, all the other `else if` statement have `this.set('dateType', dateType);` in common. Add it as a common statement.